src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -103,6 +106,18 @@ data SolverOptions = SolverOptions
   -- ^ Should the solver be allowed to defer errors by skipping constraints?
   }
 
+-- | Build a type class instance dictionary for the `IsSymbol` class for a symbol
+isSymbolDictionary :: String -> TypeClassDict
+isSymbolDictionary sym =
+  TypeClassDictionaryInScope (Left dict) [] C.IsSymbol [TypeLevelString sym] Nothing
+  where
+    inst = ObjectLiteral
+      [ ("reflectSymbol", Abs (Left (Ident C.__unused)) (Literal (StringLiteral sym)))
+      ]
+    classTy = TypeConstructor (fmap coerceProperName C.IsSymbol)
+    expr = TypedValue False (Literal inst) (TypeApp classTy (TypeLevelString sym))
```

Comment:
I wasn't sure about `TypedValue` - have I done this correctly?  I didn't work out what the boolean argument was for and I didn't see it mentioned in the doc-comment for the constructor.